src/routes/cache-hooks/guards/pre-execution.guard.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Inject,
+  Injectable,
+} from '@nestjs/common';
+import { ILoggingService, LoggingService } from '@/logging/logging.interface';
+import { formatRouteLogMessage } from '@/logging/utils';
+
+/**
+ * The PreExecutionLogGuard guard outputs a log line containing the request data.
+ */
+@Injectable()
+export class PreExecutionLogGuard implements CanActivate {
+  private readonly PRE_EXECUTION_LOGGING_DETAIL = 'pre-execution-log';
+  private readonly PRE_EXECUTION_LOGGING_STATUS = 202;
+
+  constructor(
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+  ) {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const startTimeMs: number = performance.now();
+    const httpContext = context.switchToHttp();
+    const request = httpContext.getRequest();
+
+    this.loggingService.info(
+      formatRouteLogMessage(
+        this.PRE_EXECUTION_LOGGING_STATUS,
+        request,
+        startTimeMs,
+        this.PRE_EXECUTION_LOGGING_DETAIL,
+      ),
+    );
```

Comment:
I'd choose a different format for the route. This log ends up being printed anyways by the `RouteLoggerInterceptor`. This log can represent something specific to this guard. The `request` part only has the path of the route which we know for this one (and it's already being distinguished with `PRE_EXECUTION_LOGGING_DETAIL`).

What about the following:

```typescript
canActivate(context: ExecutionContext): boolean {
  this.loggingService.info({
    type: PreExecutionLogGuard.PRE_EXECUTION_LOGGING_DETAIL,
  });
  return true;
}
```

Reply:
So by doing that, we would have a log line emitted by this guard and another log line emitted by the `RouteLoggerInterceptor`. The first would not have a route param, but the second would. 

That would let us check each line having `PreExecutionLogGuard.PRE_EXECUTION_LOGGING_DETAIL` is paired to another log line with `route: '/v1/hooks/events'`. That may work when the guard is only associated with a single route.

Reply:
I think it's a bit more consistent to log the route in the guard as well, but let me know if you think otherwise.

I've changed the format to log the path and the discrimination string in [5c9cfb7](https://github.com/safe-global/safe-client-gateway/pull/1120/commits/5c9cfb7870538e0177c956a10ef91b3e7b9b620f)